### PR TITLE
fix(home-page): fix form console warning

### PR DESCRIPTION
**Pull request by trybick**
#### Summary

- Fix "form not submitted" console warning when entering username on home page
- Shortens some repos logic in UserPage

#### Checklist

- [x] Branch is in format `TYPE/scope/description`
- [x] PR title is in semantic format `TYPE(scope): description`
- [x] Commits Messages are in semantic format `TYPE(scope): description`
- [x] Has Summary
- [x] Has Labels

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -22505,6 +22507,7 @@
       "version": "1.2.13",
       "resolved": "https://registry.npmjs.org/fsevents/-/fsevents-1.2.13.tgz",
       "integrity": "sha512-oWb1Z6mkHIskLzEJ/XWX0srkpkTQ7vaopMQkyaEIoq0fmtFVxOthb8cCxeT+p3ynTdkk/RZwbgG4brR5BeWECw==",
+      "hasInstallScript": true,
```
Comment (trybick) [inline package-lock.json:22510]: So I think upgrading my npm fixed the package lock differences. But I had this difference so I thought it made sense to commit

File: src/components/pages/HomePage.js [hunk 2]
```diff
@@ -10,7 +10,8 @@ export const HomePage = () => {
     setUsername(event.target.value);
   };
 
-  const onSubmit = () => {
+  const onSubmit = (event) => {
+    event.preventDefault();
```
Comment (trybick) [inline src/components/pages/HomePage.js:14]: fixes console warning

File: src/components/pages/UserPage.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ export const UserPage = ({ match }) => {
   const { user } = match.params;
   const [userRepos, setUserRepos] = useRecoilState(userRepoState);
   const [, setUserCommitHistory] = useRecoilState(userCommitHistoryState);
-  const { [user]: repos = [] } = userRepos;
+  const { [user]: currentUserRepos = [] } = userRepos;
```
Comment (trybick) [inline src/components/pages/UserPage.js:14]: rename to help differentiate that this is just the current user, not all repos

File: src/components/pages/UserPage.js [hunk 4]
```diff
@@ -39,10 +39,7 @@ export const UserPage = ({ match }) => {
       }),
     );
 
-    // clone previous userRepos and append new key value pair
-    const newUserRepos = Object.assign({}, userRepos);
-    Object.assign(newUserRepos, { [user]: repoData });
-    setUserRepos(newUserRepos);
+    setUserRepos({ ...userRepos, [user]: repoData });
```
Comment (davidholyko) [inline src/components/pages/UserPage.js:42]: if I remember correctly, there's a possible bug that this slows down due to the destructuring nature. Can you try looking up 20 usernames or so?

if the performance doesn't seem to drop i this change should be fine

File: src/components/pages/UserPage.js [hunk 5]
```diff
@@ -81,7 +78,7 @@ export const UserPage = ({ match }) => {
         ></input>
       </form>
 
-      <RepoList repos={repos} user={user} />
+      <RepoList repos={currentUserRepos} user={user} />
```
Comment (davidholyko) [inline src/components/pages/UserPage.js:80]: mini nit can we remove the new line above. theres also a new line in JSX for RepoList file

**Review by davidholyko — APPROVED** [review]

**Pull request merged**